concepts/observability-configurable-logging/08-revised-api.md

```diff
@@ -0,0 +1,298 @@
+
+## Considerations
+
+### Tag per pipeline mandatory
+
+Different pipelines will require different filters. By default there is only one tag like `kube.*` resulting from the central tail plugin. Every `LogPipeline` will require a dedicated `rewrite_tag` filter in order to apply some filtering, otherwise you will get indeterministic effects (the order of the filters is important).
+-> Every pipeline needs a rewrite_tag introducing a custom tag per pipeline. To generate filters later out of a `LogPreset`, the tag name needs to be known
+
+### Filesystem buffer mandatory
+
+We realized that using in-memory buffers will not decouple the pipelines well enough. If one output is down, the other output will stop working as well. It requires to enable a filesystem buffer for pipeline including size management to assure that the node filesystem gets not flooded
+-> Every pipeline needs a filesystem buffer, every output needs a storage size defaulting/validation
+
+### Match expressions should be limited
+
+Support of full match expressions in a `LogPipeline` will harm other pipelines as well as the central `tail` pipeline. People copying code snippets might have match expressions using `*`. It should be prevented.
+-> Match expressions should be defaulted/validated
+
+### Unsupported Plugins
+
+We need a clear distinction between supported/tested setups and unsupported setups. An HTTP plugin usage should be supported with any configuration while a stackdriver plugin usage is unsupported as we have no chance to cover testing. The additional resource consumption of some plugins and settings is also a relevant factor. We don't want the user to configure anything that creates a new emitter, which increases the memory usage above the amount that we introduce ourselves.
+-> Introduce a mechanism to detect usage of unsupported plugins
+
+### Dedotting support
+
+There is no filter available for opensearch-typical dedotting. That is a mandatory feature and we planned to solve it via a lua script. On the other hand, the lua filter should be an unsupported feature. So having the convinience of detotting support in the `LogPresetBinding` would result in a lua script in the `LogPipeline`.
+-> Add dedotting support as a feature to the `LogPipeline`
+
+### Meaningfulness of LogPreset
+
+It turned out that an OpenSearch integration via a HTTP plugin required for an internal SAP scenario can be much simpler then expected. A real templating is not needed and probably a plain simple pipeline config will be better suited. Also, what presets to provide, we cannot think of other scenarios which can be delivered. The binding to a secret can be done already in the pipeline itself, workload selectors could be added as well
```

Comment:
I don't understand this sentence: "Also, what presets to provide, we cannot think of other scenarios which can be delivered." Just guessing here:

```suggestion
It turned out that an OpenSearch integration with an HTTP plugin required for an internal SAP scenario can be much simpler than expected. A real templating is not needed, and probably a plain simple pipeline config will be better suited. Also, regarding presets to provide, we cannot think of other scenarios that can be delivered. The binding to a secret can be done already in the pipeline itself, workload selectors could be added as well.
```